airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -360,14 +360,19 @@ def get_cluster_state(self, clusterName: str) -> ClusterStates:
         :rtype: ClusterStates
         """
         eks_client = self.conn
+        result = None
 
         try:
-            return ClusterStates(eks_client.describe_cluster(name=clusterName).get('cluster').get('status'))
+            result = ClusterStates(eks_client.describe_cluster(name=clusterName).get('cluster').get('status'))
         except ClientError as ex:
             if ex.response.get("Error").get("Code") == "ResourceNotFoundException":
-                return ClusterStates.NONEXISTENT
+                result = ClusterStates.NONEXISTENT
```

Comment:
Oh, we are missing an `else: raise` -- that was the case that didn't have a return.